src/main/java/org/elasticsearch/index/mapper/core/AbstractFieldMapper.java

```diff
@@ -273,12 +273,6 @@ public String buildFullName(BuilderContext context) {
         }
     }
 
-    private static final ThreadLocal<List<Field>> FIELD_LIST = new ThreadLocal<List<Field>>() {
```

Comment:
OH,I see. Misread the code :( .

In that case +1 on removal , it's typically one or two values. Note how the old code initialized it with a capacity of 2 - maybe keep that?